s3/src/main/scala/monix/connect/s3/S3.scala

```diff
@@ -52,63 +58,197 @@ import software.amazon.awssdk.services.s3.model.{
-    * @param acl the canned ACL (Access Control List) to apply to the object.
-    *            if the service returns an enum value that is not available in the current SDK version, acl will return ObjectCannedACL.UNKNOWN_TO_SDK_VERSION.
-    * @param grantFullControl gives the grantee READ, READ_ACP, and WRITE_ACP permissions on the object.
-    * @param grantRead allows grantee to read the object data and its metadata.
-    * @param grantReadACP allows grantee to read the object ACL.
-    * @param grantWriteACP allows grantee to write the ACL for the applicable object.
-    * @param s3AsyncClient an implicit instance of a [[S3AsyncClient]].
-    * @param objectLockEnabledForBucket specifies whether you want S3 Object Lock to be enabled for the new bucket.
-    * @return a [[Task]] with the [[CreateBucketResponse]].
+    * @see how does the expected `.conf` file should look like
+    *      https://github.com/monix/monix-connect/blob/master/aws-auth/src/main/resources/reference.conf`
+    *
+    * @see the cats effect resource data type: https://typelevel.org/cats-effect/datatypes/resource.html
+    *
+    * @return a [[Resource]] of [[Task]] that allocates and releases [[S3]].
+    */
+  def create(): Resource[Task, S3] = {
+    Resource.make {
+      for {
+        clientConf  <- Task.fromTry(Try(AppConf.loadOrThrow))
+        asyncClient <- Task.now(AsyncClientConversions.fromMonixAwsConf(clientConf.monixAws))
+      } yield {
+        self.createUnsafe(asyncClient)
+      }
+    } { s3 => Task(s3.close()) }
+  }
+
+  /**
+    * Creates a [[Resource]] that will use the passed
+    * AWS configurations to allocate and release [[S3]].
+    * Thus, the api expects an `application.conf` file to be present
+    * in the `resources` folder.
+    *
+    * @see how does the expected `.conf` file should look like
+    *      https://github.com/monix/monix-connect/blob/master/aws-auth/src/main/resources/reference.conf`
+    *
+    * @see the cats effect resource data type: https://typelevel.org/cats-effect/datatypes/resource.html
+    *
+    * ==Example==
+    *
+    * {{{
+    *   import cats.effect.Resource
+    *   import monix.eval.Task
+    *   import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+    *   import software.amazon.awssdk.regions.Region
+    *
+    *   val defaultCredentials = DefaultCredentialsProvider.create()
+    *   val s3Resource: Resource[Task, S3] = S3.createWith(defaultCredentials, Region.AWS_GLOBAL)
+    * }}}
+    *
+    * @param credentialsProvider Strategy for loading credentials and authenticate to AWS S3
+    * @param region An Amazon Web Services region that hosts a set of Amazon services.
+    * @param endpoint The endpoint with which the SDK should communicate.
+    * @param httpClient Sets the [[SdkAsyncHttpClient]] that the SDK service client will use to make HTTP calls.
+    * @return a [[Resource]] of [[Task]] that allocates and releases [[S3]].
+    **/
+  def createWith(
```

Comment:
I'm not sure about naming, I would probably go for something like `fromConfig` (instead of `create`) and `create` (instead of `createWith`) but that's very subjective.

Reply:
one of the intention of using `create()` for creating from config was to incentive users to use it :P 
but I like your proposition!